### ISSUE TITLE
Lifx devices not showing up since recent discover refactor

### ISSUE THREAD

**Issue by catchdave** [Djelibeybi/ha-lifx-beta#16]
Since the recent upgrade, lifx devices do not show up. My system is on a rpi4. It has ethernet and wifi networks, each on separate subnets. Lifx devices are discoverable and work fine in iOS app. Lifx is only on the wifi subnet.

I installed the latest beta (`2022.06.06`) and have setup debug logs per instructions, but I don't see any clues as to why nothing is being discovered. This is what I put in my `configutaion.yaml`:
```
logger:
  default: info
  logs:
    custom_components.lifx: debug
```

Nothing stands out in my logs regarding lifx. You mentioned in a previous comment that I need to go throuigh the UI to enable it. I deleted and re-added the integration and it still doesn't work (it's actually worse now, as my lifx devices are now missing completely).

Logs:
[home-assistant.log](https://github.com/Djelibeybi/ha-lifx-beta/files/8885396/home-assistant.log)

**Comment by Djelibeybi** [comment]
So that looks like the debug logs of Home Assistant starting and then you added the LIFX integration (or enabled its configuration) and then it stops. You should have seen a whole bunch of `DEBUG` log entries for `custom_components.lifx`. 

Please add `homeassistant.setup: debug` and `homeassistant.components.lifx: debug` under `logs:` so we can see if there is a problem setting up the custom LIFX integration or if the core integration is somehow being used instead of the custom integration (for some reason):

```yaml
logger:
  default: info
  logs:
    homeassistant.setup: debug
    homeassistant.components.lifx: debug
    custom_components.lifx: debug
```

**Comment by Djelibeybi** [comment]
BTW, you should see something like this after you create the LIFX configuration on the Settings -> Integrations page:

```log
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.setup] Setting up lifx
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.setup] Setup of domain lifx took 0.0 seconds
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.setup] Setting up light
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.setup] Setup of domain light took 0.0 seconds
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.components.light] Setting up light.lifx
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.49 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.62 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.47 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.93 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.92 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.57 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.40 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.89 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.98 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.96 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.72 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.50 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.97 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.44 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.41 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.135 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.46 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.95 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.52 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.85 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.71 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.43 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.45 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Garage Ceiling Light (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new light.lifx entity: light.garage_ceiling_light
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Theatre 3R (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Theatre 2L (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Theatre 4L (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new light.lifx entity: light.theatre_3r
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new light.lifx entity: light.theatre_2l
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new light.lifx entity: light.theatre_4l
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Entrance Hall Front Door Light (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Alfresco Ceiling Light (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Theatre Table Lamp (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Test Bulb (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] First discovery response received from 192.168.254.61 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Skipping discovered LIFX Switch at 192.168.254.98 (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Laundry Ceiling Light (d0:73:d5:xx:xx:xx)
2022-06-12 17:51:16 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new light.lifx entity: light.alfresco_ceiling_light
2022-06-12 17:51:16 DEBUG (MainThread) [homeassistant.components.lifx.light] Entity created for Andrew's Office Ceiling Light (d0:73:d5:xx:xx:xx)
```

**Comment by catchdave** [comment]
Something weird is going on - I get no debug log entries for any lifx components (just info). I even set global default to debug and I got an insubordinate amount of debug logs for everything except lifx.

```
logger:
  default: debug
  logs:
    homeassistant.setup: debug
    homeassistant.components.lifx: debug
    custom_components.lifx: debug
```

**Comment by catchdave** [comment]
I can't tell why I am not getting debug logs for lifx. Here's what I can tell you:
* Running rpi4, debian 10, supervisor/hass.io installation.
* configuration.yaml has log settings like previous comment (except I reverted the default to **info**, since it was printing 10's of thousands of lines).
* I installed this beta repo in HACS and downloaded.
* I even modified `homeassistant/custom_components/lifx/light.py` and replaced all `LOGGER.debug` to `LOGGER.info` - but still don't get debug logs for lifx.
* I have restarted many times. 

In the logs, I see DEBUG level around some things, but it just won't drill down into lifx. Example extract:
```
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 DEBUG (MainThread) [homeassistant.setup] Dependency template will wait for after dependencies ['group']
2022-06-12 14:35:03 INFO (MainThread) [homeassistant.setup] Setup of domain media_source took 3.8 seconds
2022-06-12 14:35:03 INFO (MainThread) [homeassistant.setup] Setup of domain scene took 3.8 seconds
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setting up zwave_js
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setup of domain zwave_js took 0.0 seconds
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setting up lifx
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setup of domain lifx took 0.0 seconds
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setting up abode
2022-06-12 14:35:04 INFO (MainThread) [homeassistant.setup] Setup of domain abode took 0.0 seconds
```

Any more ideas to get logs? I feel like there is something fundamentally wrong, or I am missing something obvious --especially after modifying the log statements (!)

**Comment by catchdave** [comment]
OK - I have fixed the main issue. I still don't know why the debug logs don't work, but now have lifx working again.

In case this helps someone else, there is now (not sure when this change happened) an "auto" network configure in home assistant. By default it was only making ethernet available.

To fix, I simply went to `Settings>Network>Network Adapter` and unchecked **Auto configure** and instead checked the two network adapters (wifi and ethernet).
